tests/bluetooth/controller/ctrl_cte_req/src/main.c

```diff
@@ -616,6 +616,751 @@ void test_cte_req_reject_inv_ll_param_peripheral_remote(void)
+	} while (0)
+
+static bool is_instant_reached(struct ll_conn *conn, uint16_t instant)
+{
+	return ((event_counter(conn) - instant) & 0xFFFF) <= 0x7FFF;
+}
+
+static uint16_t pu_event_counter(struct ll_conn *conn)
+{
+	struct lll_conn *lll;
+	uint16_t event_counter;
+
+	/* TODO(thoh): Lazy hardcoded */
+	uint16_t lazy = 0;
+
+	/**/
+	lll = &conn->lll;
+
+	/* Calculate current event counter */
+	event_counter = lll->event_counter + lll->latency_prepare + lazy;
+
+	return event_counter;
+}
+
+static void phy_update_setup(void)
+{
+	test_setup(&conn);
+
+	/* Emulate initial conn state */
+	conn.phy_pref_rx = PHY_1M | PHY_2M | PHY_CODED;
+	conn.phy_pref_tx = PHY_1M | PHY_2M | PHY_CODED;
+	conn.lll.phy_flags = PREFER_S2_CODING;
+	conn.lll.phy_tx_time = PHY_1M;
+	conn.lll.phy_rx = PHY_1M;
+	conn.lll.phy_tx = PHY_1M;
+
+	/* Init DLE data */
+	ull_conn_default_tx_octets_set(251);
+	ull_conn_default_tx_time_set(2120);
+	ull_dle_init(&conn, PHY_1M);
+	/* Emulate different remote numbers to trigger update of eff */
+	conn.lll.dle.remote.max_tx_octets = PDU_DC_PAYLOAD_SIZE_MIN * 3;
+	conn.lll.dle.remote.max_rx_octets = PDU_DC_PAYLOAD_SIZE_MIN * 3;
+	conn.lll.dle.remote.max_tx_time = PDU_DC_MAX_US(conn.lll.dle.remote.max_tx_octets, PHY_1M);
+	conn.lll.dle.remote.max_rx_time = PDU_DC_MAX_US(conn.lll.dle.remote.max_rx_octets, PHY_1M);
+	ull_dle_update_eff(&conn);
+}
+
+static void run_local_cte_req(struct pdu_data_llctrl_cte_req *cte_req)
+{
+	struct pdu_data_llctrl_cte_rsp remote_cte_rsp = {};
+	struct node_tx *tx = NULL;
+	struct node_rx_pdu *ntf;
+
+	/* The CTE request should already be in local control procedures queue */
+
+	/* Prepare */
+	event_prepare(&conn);
+
+	/* Tx Queue should have one LL Control PDU */
+	lt_rx(LL_CTE_REQ, &conn, &tx, cte_req);
+	lt_rx_q_is_empty(&conn);
+
+	/* Rx */
+	lt_tx(LL_CTE_RSP, &conn, &remote_cte_rsp);
+
+	/* Done */
+	event_done(&conn);
+
+	/* Receive notification of sampled CTE response */
+	ut_rx_pdu(LL_CTE_RSP, &ntf, &remote_cte_rsp);
+
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+
+	/* Release tx node */
+	ull_cp_release_tx(&conn, tx);
+}
+
+void wait_for_phy_update_instant(uint8_t instant)
+{
+	/* */
+	while (!is_instant_reached(&conn, instant)) {
+		/* Prepare */
+		event_prepare(&conn);
+
+		/* Tx Queue should NOT have a LL Control PDU */
+		lt_rx_q_is_empty(&conn);
+
+		/* Done */
+		event_done(&conn);
+
+		CHECK_CURRENT_PHY_STATE(conn, PHY_1M, PREFER_S2_CODING, PHY_1M);
+
+		/* There should NOT be a host notification */
+		ut_rx_q_is_empty();
+	}
+}
+
+void check_phy_update_and_cte_req_complete(bool is_local, struct pdu_data_llctrl_cte_req *cte_req,
+					   struct pdu_data_llctrl_phy_req *phy_req,
+					   uint8_t ctx_num_at_end)
+{
+	struct pdu_data_llctrl_length_rsp length_ntf = {
+		3 * PDU_DC_PAYLOAD_SIZE_MIN,
+		PDU_DC_MAX_US(3 * PDU_DC_PAYLOAD_SIZE_MIN, phy_req->tx_phys),
+		3 * PDU_DC_PAYLOAD_SIZE_MIN,
+		PDU_DC_MAX_US(3 * PDU_DC_PAYLOAD_SIZE_MIN, phy_req->rx_phys)
```

Comment:
Here we are reusing the `3 * PDU_DC_PAYLOAD_SIZE_MIN`. 

If they need to be the same value as the above, then we should put this in a macro or a `const` value, and use that across both functions.

In any case, this struct should use a variable or a define for it, as it is using it 4 times.

Reply:
Changed to macro.